packages/google_sign_in/example/lib/main.dart

```diff
@@ -94,18 +97,45 @@ class SignInDemoState extends State<SignInDemo> {
     return null;
   }
 
-  Future<Null> _handleSignIn() async {
-    try {
-      await _googleSignIn.signIn();
-    } catch (error) {
-      print(error);
-    }
+  void _handleSignIn() {
+    _handleGoogleSignIn()
+        .then((GoogleSignInAccount user) => _afterSignIn)
+        .catchError((dynamic error) => _onSignInError);
```

Comment:
In Flutter, in general, we prefer the async/await combo to future.then.catchError. It has distinct advantages such as being more readable and selectively catching exceptions.